### PR TITLE
crypto/store/ossl_result.c: Better filtering of errors

### LINKED ISSUE #14973: GOST engine external tests failed
e.g. https://github.com/openssl/openssl/pull/14972/checks?check_run_id=2400316825

OpenSSL cmdline: /home/runner/work/openssl/openssl/apps/openssl ocsp -issuer ocspCA-gost2001_A/cacert.pem -CAfile ocspCA-gost2001_A/cacert.pem -cert U_ocsp_gost2001_A/cert.pem -signer U_ocsp_gost2001_A/cert.pem -signkey U_ocsp_gost2001_A/seckey.pem -reqout request2.der
	STDERR CONTENTS:
	40C7F5815B7F0000:error:1E08010C:DECODER routines:OSSL_DECODER_from_bio:unsupported:crypto/encode_decode/decoder_lib.c:82:No supported for the data to decode. Input type: DER, Input structure: pkcs8

Missed when did it appear first time

### PR DESCRIPTION
**Pull request by levitte**
The diverse variants of try_XXX() were filtering errors independently
of each other.  It's better done in ossl_store_handle_load_result()
itself, where we have control over the overall success and failure of
the attempts.

Fixes #14973

### CODE REVIEW

File: providers/implementations/storemgmt/file_store_der2obj.c [hunk 1]
```diff
@@ -87,29 +87,18 @@ static int der2obj_decode(void *provctx, OSSL_CORE_BIO *cin, int selection,
...
-        if (ERR_GET_LIB(err) == ERR_LIB_ASN1
-            && (ERR_GET_REASON(err) == ASN1_R_HEADER_TOO_LONG
-                || ERR_GET_REASON(err) == ASN1_R_UNSUPPORTED_TYPE
-                || ERR_GET_REASON(err) == ERR_R_NESTED_ASN1_ERROR
-                || ERR_GET_REASON(err) == ASN1_R_NOT_ENOUGH_DATA)) {
-            ERR_pop_to_mark();
-        } else {
-            ERR_clear_last_mark();
-            goto end;
-        }
```
Comment (beldmit) [inline providers/implementations/storemgmt/file_store_der2obj.c:112]: I have a naive question. Didn't we hide some real errors in such a way (e.g. malformed key/unsupported format variation)?

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by beldmit** [comment]
As #14973 is labeled as urgent, shouldn't we merge this PR?

**Comment by levitte** [comment]
Yeah, I'll merge it now

**Comment by levitte** [comment]
Merged

e36a4dc476448a2ef212d774be48ce38ea6eb6df EVP: evp_keymgmt_util_try_import() should clean up on failed import
521a0bf6a11c4cdaef331934e93581d06ce834e1 crypto/store/ossl_result.c: Better filtering of errors
3d80b5e611f112fd004a4320cb5ecce93c73b7d4 STORE: Simplify error filtering in der2obj_decode()
a70936a8453a307992820f2a9d3e252f6c4f9ad6 TEST: correct test/recipes/30-test_evp_data/evppkey_ecdh.txt

**Pull request closed without merging**